src/prpy/util.py

```diff
@@ -693,3 +693,44 @@ def FindCatkinResource(package, relative_path):
     else:
         raise IOError('Loading resource "{:s}" failed.'.format(
                       relative_path))
+
+
+def IsAtTrajectoryStart(robot, trajectory):
+    """
+    Check if robot's active DOFs match the start configuration of a trajectory.
+
+    Examines the current active DOF values of the specified robot and compares
+    these values to the first waypoint of the specified trajectory.  If every
+    DOF value differs by less than the DOF resolution of the specified
+    joint/axis, then the function returns True.  Otherwise, it returns False.
+
+    @param robot: the robot whose active DOFs will be checked
+    @param trajectory: the trajectory whose start configuration will be checked
+    @returns: True if the robot's active DOFs match the given trajectory
+              False if one or more active DOFs differ by DOF resolution
+    """
+    # Get current active configuration of robot.
+    dof_indices = robot.GetActiveDOFIndices()
+    dof_values = robot.GetActiveDOFValues()
+    dof_resolutions = robot.GetActiveDOFResolutions()
```

Comment:
Note also that the documentation for OpenRAVE [0] suggests that the notion of active DOFs is meant only for planners (not executors, etc):

> Set the joint indices and affine transformation dofs that the planner should use.

[0] http://openrave.org/docs/latest_stable/coreapihtml/classOpenRAVE_1_1RobotBase.html#a818d340885e3a5b0372c7bae025430af